src/jaegertracing/TracerFactory.cpp

```diff
@@ -24,10 +24,7 @@ opentracing::expected<std::shared_ptr<opentracing::Tracer>>
 TracerFactory::MakeTracer(const char* configuration,
                           std::string& errorMessage) const noexcept try {
 #ifndef JAEGERTRACING_WITH_YAML_CPP
-    errorMessage =
-        "Failed to construct tracer: Jaeger was not build with yaml support.";
-    return opentracing::make_unexpected(
-        std::make_error_code(std::errc::not_supported));
+    auto tracerConfig = jaegertracing::Config();
```

Comment:
this feels unintuitive - the `configuration` is silently ignored in favor of default config?  Maybe it should at least have a null check and still error our if not null?

Reply:
also, does C++ have some standard way to document functions? This one could use it.

Reply:
> this feels unintuitive - the configuration is silently ignored in favor of default config?

I reverted now to the old behaviour of erroring out when no config is provided as it feels out of scope of this PR to change that.

> does C++ have some standard way to document functions?

I haven't done C++ in a long time so I would need on your knowledge, but looking at the other files I don't see any documentation.